huntsman/camera/pyro.py

```diff
@@ -177,73 +185,51 @@ def take_exposure(self,
+            dark (bool, optional): Exposure is a dark frame, default False. On cameras that support
+                taking dark frames internally (by not opening a mechanical shutter) this will be
+                done, for other cameras the light must be blocked by some other means. In either
+                case setting dark to True will cause the `IMAGETYP` FITS header keyword to have
+                value 'Dark Frame' instead of 'Light Frame'. Set dark to None to disable the
+                `IMAGETYP` keyword entirely.
             blocking (bool, optional): If False (default) returns immediately after starting
                 the exposure, if True will block until it completes.
 
         Returns:
-            threading.Event: Event that will be set when exposure is complete
+            threading.Event: Event that will be set when exposure is complete.
 
         """
-        assert self.is_connected, self.logger.error("Camera must be connected for take_exposure!")
-
-        assert filename is not None, self.logger.warning("Must pass filename for take_exposure")
-
-        if not self._exposure_event.is_set():
-            msg = "Attempt to take exposure on {} while one already in progress.".format(self)
-            raise error.PanError(msg)
-
-        # Clear event now to prevent any other exposures starting before this one is finished.
-        self._exposure_event.clear()
-
-        # Make sure proxy is in async mode
-        Pyro4.asyncproxy(self._proxy, asynchronous=True)
-
         # Start the exposure
-        self.logger.debug(f'Taking {seconds} second exposure on {self.name}: {filename}')
+        self.logger.debug(f'Taking {seconds} second exposure on {self}: {filename}')
 
         # Remote method call to start the exposure
-        exposure_result = self._proxy.take_exposure(seconds=seconds,
-                                                    filename=filename,
-                                                    dark=bool(dark),
-                                                    *args,
-                                                    **kwargs)
-
-        # Start a thread that will set an event once exposure has completed
-        exposure_thread = Timer(interval=get_quantity_value(seconds, u.s) + self.readout_time,
-                                function=self._async_wait,
-                                args=(exposure_result,
-                                      'exposure',
-                                      self._exposure_event,
-                                      self._timeout))
-        exposure_thread.start()
+        self._proxy.take_exposure(seconds=seconds,
+                                  filename=filename,
+                                  dark=bool(dark),
+                                  *args,
+                                  **kwargs)
 
+        max_wait = get_quantity_value(seconds, u.second) + self.readout_time + self._timeout
```

Comment:
Shouldn't it be `max((seconds + readout), self._timeout)`?  That is, the length of exposure and readout (and maybe some padding) or the timeout?

Reply:
Hmmmm, we probably need to clarify the role of the `Camera` class `timeout` attribute. In the POCS `Camera` base class `take_exposure` starts a `threading.Timer` with duration of `seconds`, which then starts the polling loop which runs for a maximum period of `timeout` seconds. So in that case the maximum elapsed time to wait for an exposure is `seconds + timeout`, the `readout_time` property is totally ignored (!), and `timeout` should be set to the camera's nominal readout time plus a little padding.

Here I'm using the `timeout` attribute of the Pyro `Camera` (note, it will in general be different from the corresponding remote camera's `timeout`) as the padding and so set the maximum elapsed time to wait for the exposure to `seconds + readout_time + timeout`. This makes more sense to me than what POCS is currently doing, but we should be consistent here.

Maybe it can be cleared up with renaming the current `._timeout` attribute to something like `._timeout_padding` and then the above line becomes `timeout = get_quantity_value(seconds, u.second) + self.readout_time + self.timeout_padding`.